spec/helpers/activity_helper_spec.rb

```diff
@@ -62,6 +65,137 @@
     end
   end
 
+  describe "#link_to_add_comment" do
+    context "when activity is a programme" do
+      context "when user is authorised to add programme comments" do
+        it "returns button to add comment" do
+          authorise_creating_comments(commentable_type: :programme_activity, authorise: true)
+
+          expect(helper.link_to_add_comment(activity: programme_activity, report: nil)).to include new_activity_comment_path(programme_activity)
+          expect(helper.link_to_add_comment(activity: programme_activity, report: nil)).to include t("page_content.comment.add")
+        end
+      end
+
+      context "when user is unauthorised to add programme comments" do
+        it "returns nil" do
+          authorise_creating_comments(commentable_type: :programme_activity, authorise: false)
+          authorise_creating_comments(commentable_type: :project_activity, authorise: true)
+
+          expect(helper.link_to_add_comment(activity: programme_activity, report: nil)).to be_nil
+        end
+      end
+    end
+
+    context "when activity is a project" do
+      context "when user is authorised to add project comments" do
+        context "when a report exists" do
+          it "returns button to add comment" do
+            authorise_creating_comments(commentable_type: :programme_activity, authorise: false)
```

Comment:
again, if we move this up to the `when user is authorised to add project comments` context, we don't need to repeat it in the test below.